business/services.go

```diff
@@ -120,19 +120,25 @@ func (in *SvcService) GetService(namespace, service, interval string, queryTime
 	var nsmtls models.MTLSStatus
 
 	wg := sync.WaitGroup{}
-	wg.Add(9)
+	wg.Add(7)
 	errChan := make(chan error, 6)
 
 	labelsSelector := labels.Set(svc.Spec.Selector).String()
+	// If service doesn't have any selector, we can't know which are the pods and workloads applying.
+	if labelsSelector != "" {
+		wg.Add(2)
+	}
 
-	go func() {
-		defer wg.Done()
-		var err2 error
-		pods, err2 = in.k8s.GetPods(namespace, labelsSelector)
-		if err2 != nil {
-			errChan <- err2
-		}
-	}()
+	if labelsSelector != "" {
```

Comment:
Minor comment, perhaps the goroutine for GetPods and fetchWorkloads can be under same check of labelSelector.
But it's Ok.